src/pages/resultsView/ResultsViewPage.tsx

```diff
@@ -39,12 +48,22 @@ function initStore() {
         getBrowserWindow().currentQueryStore = createQueryStore();
     }
 
-    const reaction1 = reaction(
+    let lastQuery:any;
+
+    const queryReactionDisposer = reaction(
         () => {
             return getBrowserWindow().globalStores.routing.query
         },
         (query) => {
 
+            // escape from this if queryies are deeply equal
+            // TODO: see if we can figure out why query is getting changed and
+            // if there's any way to do shallow equality check to avoid this expensive operation
+            if (_.isEqual(lastQuery, query)) {
```

Comment:
you can use ExtendedRouterStore::queryHash :)

but I just realized that if you're having any weirdness with the insertion order of the query object, then you may have to update that method to use some package that does a "stable stringify" eg https://www.npmjs.com/package/json-stable-stringify

Reply:
I don't think insertion order matters because we are doing deep comparison of object.  nothing stringified